README.md

```diff
@@ -0,0 +1,123 @@
+# API Specification Version Maven Plugin
+
+The API Specification Version Plugin provides goals to set properties for manifest file OSGI headers and to verify consistence of specification, API and API implementation package names and versions.
+
+**Goals:**
+* `set-spec-properties` (validate phase)
+* `check-module` (package phase)
+
+## Configuration properties
+
+| Property | Type | Default<br/>Value | Description |
+| --- | --- | --- | --- |
+| specMode | `jakarta`&vert;`javax` | `jakarta` | `jakarta`: Jakarta EE projects mode<br/>`javaee`: legacy mode for java.net projects (deprecated) |
+| ignoreErrors | boolean | false | whether `check-module` target should just print warnings or fail the build on errors |
+| spec.nonFinal | boolean | false | `true` for snapshot build, `false` for release build
+| spec.jarType | `api`&vert;`impl`| `api` | `api`: specification with a separate API jar file <br/>`impl`: specification with a standalone API jar file |
+| spec.apiPackage | java package | | specification API package prefix |
+| spec.specVersion | version | | specification version |
+| spec.newSpecVersion | version | | specification version for snapshot builds (non final API) |
+| spec.specImplVersion | version | | specification implementation version |
+| spec.newImplVersion | version | | specification implementation version for snapshot builds (non final API) |
+
+## set-spec-properties Goal
+
+Sets properties for manifest file OSGI headers.
+
+Configuration properties always required:
+* `spec.specVersion`
+* `spec.apiPackage`
+
+### Properties Mapping for `api` jarType
+
+| Property | OSGI Header | Source |
+| --- | --- | --- |
+| `spec.bundle.symbolic-name` | Bundle-SymbolicName | `spec.apiPackage` |
+| `spec.bundle.spec.version`| BundleSpecVersion | `spec.specVersion` |
+| `spec.bundle.version`| Bundle-Version | `spec.specImplVersion` |
+| `spec.extension.name`| Extension-Name | `spec.apiPackage` |
+| `spec.specification.version`| Specification-Version | `spec.specVersion` |
+| `spec.implementation.version`| Implementation-Version | `spec.specImplVersion` |
+
+### Properties Mapping for `impl` jarType
+
+| Property | OSGI Header | Source |
+| --- | --- | --- |
+| `spec.bundle.symbolic-name` | Bundle-SymbolicName | `spec.implNamespace '.' spec.apiPackage` |
+| `spec.bundle.spec.version`| BundleSpecVersion | `spec.specVersion` |
+| `spec.bundle.version`| Bundle-Version | `spec.implVersion` |
+| `spec.extension.name`| Extension-Name | `spec.apiPackage` |
+| `spec.specification.version`| Specification-Version | `spec.specVersion` |
+| `spec.implementation.version`| Implementation-Version | `project.version`?? |
+
+## check-module Goal
+
+Validates maven properties consistency.
+
+Properties always required:
```

Comment:
Not sure we really need to state the rules that are enforced, a link to the document may be enough.
If we still want to do this and it should be a little more formal.

Reply:
It makes sense to me to keep those lists here. People may not know what spec parameters are required by this goal and why.
But give me URL of Bills document first, maybe it will say ewerything we need.

Reply:
https://wiki.eclipse.org/JakartaEE_Maven_Versioning_Rules

Reply:
Lists removed. This document seems to be enough.